tests/test_log.py

```diff
@@ -57,3 +61,121 @@ def test_logger_config():
     config = logger_config._persistent_data.load_config()
     assert config["logging"] == {'use_python_logger': False, 'enable_file': False,
                                  'channels': {'general': {'level': 'off'}, 'resources': {'level': 'off'}, 'user': {'level': 'warn'}}}
+
+
+def _get_last_dict_logged(mock_info_args):
+    return json.loads(mock_info_args)
+
+
+SOURCE = 'source'
+NOISE = 'noise'
+LAYER = 'layer'
+N = 'n'
+M = 'm'
+BACKEND = 'backend'
+METHOD = 'method'
+
+
+@patch.object(pcvl.logger, "info")
+def test_log_resources(mock_info, requests_mock):
+    pcvl.logger.set_level(pcvl.logging.level.info, pcvl.logging.channel.resources)
```

Comment:
Perhaps you could split the tests up for different objects (referring to Processor, simulator, etc.), to make it more readable.